File: src/BestTimeToBuyAndSellStockII.java
public class BestTimeToBuyAndSellStockII {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) {
            return 0;
        }

        int sellPrice = prices[0];
        int buyPrice = prices[0];
        int profit = 0;

        for (int index = 1 ; index <prices.length ; index++) {
            if (prices[index] < sellPrice) {
                profit += sellPrice - buyPrice;
                sellPrice = buyPrice = prices[index];
            } else {
                sellPrice = prices[index];
            }
        }
        profit += sellPrice - buyPrice;

        return profit;
    }
}

File: python/best_time_to_buy_and_sell_stock_ii.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy_price, sell_price, profit = prices[0], prices[0], 0
        for price in prices:
            if price < sell_price:
                profit += sell_price - buy_price
                buy_price = price
                sell_price = price
            else:
                sell_price = price
        profit += sell_price - buy_price
        return profit
